test/local/main.go

```diff
@@ -0,0 +1,172 @@
+// Copyright 2017 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package main is used for local testing of Istio proxy mesh
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"os/signal"
+	"syscall"
+
+	"istio.io/pilot/adapter/config/memory"
+	"istio.io/pilot/model"
+	"istio.io/pilot/proxy"
+	"istio.io/pilot/proxy/envoy"
+	"istio.io/pilot/test/server"
+)
+
+const (
+	localhost = "127.0.0.1"
+	http      = "http"
+	grpc      = "grpc"
+)
+
+var (
+	baseHTTPPort  int
+	baseGRPCPort  int
+	n             int
+	host          string
+	discoveryPort int
+	configpath    string
+	binarypath    string
+	service       *model.Service
+)
+
+func init() {
+	flag.IntVar(&baseHTTPPort, http, 9000, "Base HTTP/1.1 port")
+	flag.IntVar(&baseGRPCPort, grpc, 9100, "Base gRPC port")
+	flag.IntVar(&n, "n", 10, "Number of instances")
```

Comment:
Qualify what instances is this about. For example:
"Number of proxy instances" or "Envoy instance" or "app instances"